java-frontend/src/main/java/org/sonar/java/se/constraint/ConstraintManager.java

```diff
@@ -153,6 +157,14 @@ private SymbolicValue createIdentifierSymbolicValue(IdentifierTree identifier) {
     }
     return createDefaultSymbolicValue();
   }
+  
+  private SymbolicValue createVariableSymbolicValue(VariableTree identifier) {
+    final Type type = identifier.symbol().type();
+    if (type != null && type.is("java.lang.Boolean")) {
```

Comment:
I'm really not sure about this `null` test here, I think it's impossible. At worst, an unknown symbol should return an unknown type., and so your test can probably simply be `if (type.is("java.lang.Boolean")) { ...`